runtime/doc/lua-plugin.txt

```diff
@@ -0,0 +1,290 @@
+          -- Moving computation into a separate function allows using the
+          -- "early return" pattern
+          return i
+        end
+      end
+      return 0
+    end
+
+    return M
+<
+
+Now there are two main approaches on how this can be used in main module:
+eager loading and lazy loading.
+
+Eager approach is about sourcing "helper" module during source "main" module:
+>lua
+    -- lua/no-trailing/init.lua
+
+    local M = {}
+
+    -- This will make initial source of 'no-trailing.utils' module
+    -- during initial source of this module
+    local utils = require('no-trailing.utils')
+
+    M.trim = function(buf_id, opts)
+      local ok, is_buf_valid = pcall(vim.api.nvim_buf_is_valid, buf_id)
+      assert(ok and is_buf_valid, '`buf_id` is not a valid buffer identifier')
+
+      opts = vim.tbl_deep_extend('force', { empty = false }, opts or {})
+      assert(type(opts.empty) == 'boolean', '`opts.empty` is not boolean')
+
+      local start_line = utils.get_last_trailing(buf_id, opts.empty)
+      vim.api.nvim_buf_set_lines(buf_id, start_line, -1, false, {})
+    end
+
+    return M
+<
+Lazy approach is about sourcing "helper" module during function execution:
+>lua
+    -- lua/no-trailing/init.lua
+
+    local M = {}
+
+    M.trim = function(buf_id, opts)
+      local ok, is_buf_valid = pcall(vim.api.nvim_buf_is_valid, buf_id)
+      assert(ok and is_buf_valid, '`buf_id` is not a valid buffer identifier')
+
+      opts = vim.tbl_deep_extend('force', { empty = false }, opts or {})
+      assert(type(opts.empty) == 'boolean', '`opts.empty` is not boolean')
+
+      -- This will make initial source of 'no-trailing.utils' module
+      -- during first function execution
+      local utils = require('no-trailing.utils')
+      local start_line = utils.get_last_trailing(buf_id, opts.empty)
+      vim.api.nvim_buf_set_lines(buf_id, start_line, -1, false, {})
+    end
+
+    return M
+<
+Both eager and lazy loading Lua modules have pros and cons:
+
+• Eager loading leads to spending more time during initial source of main
+  module, which is often during startup.
+  Lazy loading spends that extra time only when it is needed.
+  This rarely matters for small modules which only consists from function
+  definitions, but can be significant for modules which do something time
+  consuming during their initial source or in case of large amount of modules.
+
+• Eager loading results in a more structured code with less lines.
+
+• Lazy loading results in a constant calls of `require()` during each function
+  execution. Although it caches its results, the time it spends to search the
+  cache is not strictly zero, which leads to a very small overhead (usually
+  order of fractions of a microsecond). This might matter for frequently
+  called functions.
+
+Usually both approaches are fine. Using eager loading and preferring lazy
+loading for expensive to source modules is usually a good idea.
```

Comment:
Like, I don't think this is wrong but this is just too much. I just want something quick and easy to get started, not a deepdive into everything I might possible use. If both approaches are *usually* fine then we could just pick one and call it a day. More information is not better.